core/src/AzureMcp.Core/Extensions/OpenTelemetryExtensions.cs

```diff
@@ -39,11 +39,14 @@ public static void ConfigureOpenTelemetry(this IServiceCollection services)
 
                 if (options.IsTelemetryEnabled)
                 {
-                    var address = NetworkInterface.GetAllNetworkInterfaces().Where(x => x.OperationalStatus == OperationalStatus.Up)
+                    var address = NetworkInterface.GetAllNetworkInterfaces()
+                        .Where(x => x.OperationalStatus == OperationalStatus.Up && x.NetworkInterfaceType != NetworkInterfaceType.Loopback)
                         .Select(x => x.GetPhysicalAddress().ToString())
-                        .FirstOrDefault(string.Empty);
+                        .FirstOrDefault(x => !string.IsNullOrEmpty(x));
```

Comment:
This will supply the default "N/A" if you want it to:

```suggestion
                        .Select(Sha256Helper.GetHashedValue)
                        .FirstOrDefault(x => !string.IsNullOrEmpty(x), "N/A");
```

Reply:
I originally had that.. but I don't want the SHA-256 of "N/A"

Reply:
Sha256Helper.GetHashedValue handles null and empty well, so, in full:

```cs
    var address = NetworkInterface.GetAllNetworkInterfaces()
        .Where(x => x.OperationalStatus == OperationalStatus.Up && x.NetworkInterfaceType != NetworkInterfaceType.Loopback)
        .Select(x => Sha256Helper.GetHashedValue(x.GetPhysicalAddress().ToString()))
        .FirstOrDefault(x => !string.IsNullOrEmpty(x), "N/A");
                        
    options.MacAddressHash = address;  
```